cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -702,6 +703,100 @@ func TestValidateFeatureGates(t *testing.T) {
 }
 
 func TestValidateIgnorePreflightErrors(t *testing.T) {
+	var tests = []struct {
+		ignorePreflightErrorsFromCLI        []string
+		ignorePreflightErrorsFromConfigFile []string
+		expectedSet                         sets.String
+		expectedError                       bool
+	}{
+		{ // empty lists in CLI and config file
+			[]string{},
+			[]string{},
+			sets.NewString(),
+			false,
+		},
+		{ // empty list in CLI only
+			[]string{},
+			[]string{"a"},
+			sets.NewString("a"),
+			false,
+		},
+		{ // empty list in config file only
+			[]string{"a"},
+			[]string{},
+			sets.NewString("a"),
+			false,
+		},
+		{ // no duplicates, no overlap
+			[]string{"a", "b"},
+			[]string{"c", "d"},
+			sets.NewString("a", "b", "c", "d"),
+			false,
+		},
+		{ // some duplicates, with some overlapping duplicates
+			[]string{"a", "b", "a"},
+			[]string{"c", "b"},
+			sets.NewString("a", "b", "c"),
+			false,
+		},
+		{ // non-duplicate, but 'all' present together with individual checks in CLI
+			[]string{"a", "b", "all"},
+			[]string{},
+			sets.NewString(),
+			true,
+		},
+		{ // non-duplicate, but 'all' present in config file, which is forbidden
+			[]string{"a", "b"},
+			[]string{"all"},
+			sets.NewString(),
+			true,
+		},
+		{ // skip all checks
+			[]string{"all"},
+			[]string{"a", "b"},
+			sets.NewString("all"),
```

Comment:
not correct expected result. should be error. 'all' should not be mixed with anything else.

Reply:
My understanding was that **if** `all` is provided via the CLI, it should override the config file.
Is that not what we'd like?

Reply:
This forces to limit usage of 'all' as much as possible. Easier to add any individual check, rather than trying to blindly ignoring all possible errors from checks without thinking. We seen a lot of 'guides' in the internet where they all started with 'kubeadm .... --skip-preflight-checks' and then trying to deal with problems that were actually reported by preflight checks as errors/warnings.
So, morale for admins: if you really know what you're doing, think again, and if still think that error must be ignored - add individual error to the list. If you copy&pasting from somewhere 'all', think 3 times, are you really want to do that.